aten/src/ATen/native/Normalization.cpp

```diff
@@ -491,10 +491,27 @@ Tensor batch_norm(
     auto out = input.clone();
     if (weight.defined()) out = out * weight[0];
     if (bias.defined()) out = out + bias[0];
-    return out; 
+    return out;
   }
-  return std::get<0>(at::_batch_norm_impl_index(input, weight, bias, running_mean, running_var,
-                                                training, momentum, eps, cudnn_enabled));
+  auto result = at::_batch_norm_impl_index(
+      input,
+      weight,
+      bias,
+      running_mean,
+      running_var,
+      training,
+      momentum,
+      eps,
+      cudnn_enabled);
+  // Not all implementations of batch norm do inplace modifications of mean and var #33867
+  if (running_mean.defined()) {
```

Comment:
Returned tuple has a following structure:
```
return std::tuple<Tensor, Tensor, Tensor, Tensor>{output_t, save_mean, save_var, reserve};
```
save_mean and save_var are not running_mean/running_var, so when you are getting 1st and 2nd elements of the returned tuple, you don't get what you think you are getting,

Reply:
Changed